File: app/src/main/java/com/ladsoft/bakingapp/manager/SessionManager.java
package com.ladsoft.bakingapp.manager;

import android.content.Context;
import android.content.SharedPreferences;

public class SessionManager {
    private static final String SHARED_PREFERENCES_FILENAME = "preferences";
    private static final String LAST_SELECTED_RECEIPT_ID = "last_selected_receipt_id";

    private final SharedPreferences sharedPreferences;
    private final SharedPreferences.Editor sharedPreferencesEditor;
    public SessionManager(Context context) {
        this.sharedPreferences = context.getSharedPreferences(SHARED_PREFERENCES_FILENAME, Context.MODE_PRIVATE);
        this.sharedPreferencesEditor = sharedPreferences.edit();
    }

    public long getLastSelectedReceiptId() {
        return sharedPreferences.getLong(LAST_SELECTED_RECEIPT_ID, 0L);
    }

    public void setLastSelectedReceiptId(long receiptId) {
        sharedPreferencesEditor.putLong(LAST_SELECTED_RECEIPT_ID, receiptId).commit();
    }

}

File: app/src/main/java/com/ladsoft/bakingapp/manager/SessionManager.kt
package com.ladsoft.bakingapp.manager

import android.content.Context
import android.content.SharedPreferences

class SessionManager(val context : Context) {
    private val SHARED_PREFERENCES_FILENAME = "preferences"
    private val LAST_SELECTED_RECEIPT_ID = "last_selected_receipt_id"

    private val sharedPreferences: SharedPreferences
    private val sharedPreferencesEditor: SharedPreferences.Editor

    init {
        sharedPreferences = context.getSharedPreferences(SHARED_PREFERENCES_FILENAME, Context.MODE_PRIVATE)
        sharedPreferencesEditor = sharedPreferences.edit()
    }

    var lastSelectedReceiptId = {
        sharedPreferences.getLong(LAST_SELECTED_RECEIPT_ID, 0L)
    }

    fun setLastSelectedReceiptId(receiptId: Long) {
        sharedPreferencesEditor.putLong(LAST_SELECTED_RECEIPT_ID, receiptId).commit()
    }


}